File: Quiz application.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Function to show the quiz questions and calculate the score
void startQuiz() {
    int score = 0;
    char answer;
    
    printf("Welcome to the Quiz!\n\n");

    // Question 1
    printf("1. What is the capital of France?\n");
    printf("a) Paris\nb) London\nc) Berlin\nd) Rome\n");
    printf("Enter your answer (a/b/c/d): ");
    scanf(" %c", &answer);
    if (answer == 'a') {
        score++;
        printf("Correct!\n\n");
    } else {
        printf("Wrong! The correct answer is a) Paris\n\n");
    }

    // Question 2
    printf("2. Who wrote the Harry Potter series?\n");
    printf("a) J.K. Rowling\nb) J.R.R. Tolkien\nc) George R.R. Martin\nd) Agatha Christie\n");
    printf("Enter your answer (a/b/c/d): ");
    scanf(" %c", &answer);
    if (answer == 'a') {
        score++;
        printf("Correct!\n\n");
    } else {
        printf("Wrong! The correct answer is a) J.K. Rowling\n\n");
    }

    // Question 3
    printf("3. Which planet is known as the Red Planet?\n");
    printf("a) Earth\nb) Venus\nc) Mars\nd) Jupiter\n");
    printf("Enter your answer (a/b/c/d): ");
    scanf(" %c", &answer);
    if (answer == 'c') {
        score++;
        printf("Correct!\n\n");
    } else {
        printf("Wrong! The correct answer is c) Mars\n\n");
    }

    // Question 4
    printf("4. What is the largest mammal?\n");
    printf("a) Elephant\nb) Blue Whale\nc) Giraffe\nd) Polar Bear\n");
    printf("Enter your answer (a/b/c/d): ");
    scanf(" %c", &answer);
    if (answer == 'b') {
        score++;
        printf("Correct!\n\n");
    } else {
        printf("Wrong! The correct answer is b) Blue Whale\n\n");
    }

    // Question 5
    printf("5. Who painted the Mona Lisa?\n");
    printf("a) Leonardo da Vinci\nb) Pablo Picasso\nc) Vincent van Gogh\nd) Michelangelo\n");
    printf("Enter your answer (a/b/c/d): ");
    scanf(" %c", &answer);
    if (answer == 'a') {
        score++;
        printf("Correct!\n\n");
    } else {
        printf("Wrong! The correct answer is a) Leonardo da Vinci\n\n");
    }

    // Display final score
    printf("You scored %d out of 5.\n", score);
}

// Main function
int main() {
    int choice;

    // Main menu
    do {
        printf("===== Quiz Application =====\n");
        printf("1. Start Quiz\n");
        printf("2. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                startQuiz();
                break;
            case 2:
                printf("Thank you for playing!\n");
                break;
            default:
                printf("Invalid choice. Please try again.\n");
        }

    } while (choice != 2);

    return 0;
}

File: Quiz application.py
"""
Quiz Application in Python
"""

class Quiz:
    def __init__(self, questions):
        self.questions = questions
        self.score = 0

    def start(self):
        print("Welcome to the Quiz!")
        for idx, (question, options, correct_option) in enumerate(self.questions):
            print(f"\nQuestion {idx + 1}: {question}")
            for i, option in enumerate(options, start=1):
                print(f"{i}. {option}")
            try:
                answer = int(input("Enter your choice (1/2/3/4): "))
                if answer == correct_option:
                    print("Correct!")
                    self.score += 1
                else:
                    print(f"Wrong! The correct answer was: {options[correct_option - 1]}")
            except ValueError:
                print("Invalid input! Skipping this question.")

        print(f"\nQuiz Finished! Your score is {self.score}/{len(self.questions)}")

# Define questions as tuples: (question, [options], correct_option_index)
quiz_questions = [
    ("What is the capital of France?", ["Berlin", "Madrid", "Paris", "Rome"], 3),
    ("Which programming language is known as the 'Swiss Army Knife'?", ["Python", "C++", "Java", "Ruby"], 1),
    ("Who wrote 'To Kill a Mockingbird'?", ["Harper Lee", "J.K. Rowling", "Mark Twain", "Ernest Hemingway"], 1),
    ("What is the smallest prime number?", ["0", "1", "2", "3"], 3)
]

# Create a Quiz instance and start the quiz
if __name__ == "__main__":
    quiz = Quiz(quiz_questions)
    quiz.start()
